webui/tests/e2e-cluster/run-server.py

```diff
@@ -0,0 +1,125 @@
+import multiprocessing as mp
+import socket
+import subprocess
+import sys
+import time
+
+# While we could use something like requests (or any other 3rd-party module),
+# this script aims to work with the default Python 3.6+.
+
+CLEAR = "\033[39m"
+MAGENTA = "\033[95m"
+BLUE = "\033[94m"
+
+DB_PROT = 5433
+MASTER_PORT = 8081
+
+
+def kill_process(name, process):
+    if process is not None and process.is_alive():
+        try:
+            process.kill()
+        except Exception:
+            print(f"failed to kill process: {name}")
+
+
+def wait_for_server(port, host="localhost", timeout=5.0):
+    for _ in range(100):
+        try:
+            with socket.create_connection((host, port), timeout=timeout):
+                return
+        except OSError:
+            time.sleep(1)
+    print(f"Timed out waiting for the {host}:{port}.")
+
+
+def proc(name, cmd, logs_handler=lambda x: x):
+    def func():
+        with subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+        ) as p:
+            try:
+                for line in p.stdout:
+                    print(logs_handler(line.decode("utf8")), end="", flush=True)
+            except KeyboardInterrupt:
+                print(f"Killing Log stream for {name}")
+
+    return mp.Process(target=func, daemon=True)
+
+
+def tail_db_logs():
+    return proc("database-logs", ["docker-compose", "logs", "-f"])
+
+
+def run_master():
+    # TODO make the port configurable with a default
+    return proc(
+        "master",
+        # TODO use path lib
+        ["../../../master/build/determined-master", "--config-file", "master.yaml"],
+        logs_handler=lambda line: f"{MAGENTA}determined-master  |{CLEAR} {line}",
+    )
+
+
+def run_agent():
+    # TODO connect to the set master
+    container_master_host = "host.docker.internal" if sys.platform == "darwin" else ""
```

Comment:
yay for mac support! :D

Reply:
: ) I think windows could be there too. apparently it's only the Linux docker client that doesn't have that dns name set.